pkg/networkservice/common/excludedprefixes/server_test.go

```diff
@@ -0,0 +1,143 @@
+
+	configPath := path.Join(os.TempDir(), "excluded_prefixes.yaml")
+	f, err := os.Create(configPath)
+	require.Nil(t, err)
+	defer func() { _ = os.Remove(configPath) }()
+	_, err = f.WriteString(testConfig)
+	require.Nil(t, err)
+	err = f.Close()
+	require.Nil(t, err)
+
+	chain := next.NewNetworkServiceServer(NewServerFromPath(configPath), checkrequest.NewServer(t, func(t *testing.T, request *networkservice.NetworkServiceRequest) {
+		require.Equal(t, request.Connection.Context.IpContext.ExcludedPrefixes, prefixes)
+	}))
+	req := &networkservice.NetworkServiceRequest{
+		Connection: &networkservice.Connection{
+			Context: &networkservice.ConnectionContext{},
+		},
+	}
+
+	_, _ = chain.Request(context.Background(), req)
+}
+
+func TestCheckReloadedPrefixes(t *testing.T) {
+	var prefixesUpdated = &struct {
+		sync.RWMutex
+		b bool
+	}{}
+	prefixes := []string{"172.16.1.0/24", "10.32.0.0/12", "10.96.0.0/12"}
+	testConfig := strings.Join(append([]string{"prefixes:"}, prefixes...), "\n- ")
+
+	configPath := path.Join(os.TempDir(), "excluded_prefixes.yaml")
+	f, err := os.Create(configPath)
+	require.Nil(t, err)
+	defer func() { _ = os.Remove(configPath) }()
+	err = f.Close()
+	require.Nil(t, err)
+
+	chain := next.NewNetworkServiceServer(NewServerFromPath(configPath), checkrequest.NewServer(t, func(t *testing.T, request *networkservice.NetworkServiceRequest) {
+		if reflect.DeepEqual(request.Connection.Context.IpContext.ExcludedPrefixes, prefixes) {
+			prefixesUpdated.Lock()
+			prefixesUpdated.b = true
+			prefixesUpdated.Unlock()
+		}
+	}))
+	req := &networkservice.NetworkServiceRequest{
+		Connection: &networkservice.Connection{
+			Context: &networkservice.ConnectionContext{},
+		},
+	}
+
+	err = ioutil.WriteFile(configPath, []byte(testConfig), 0600)
+	require.Nil(t, err)
+
+	require.Eventually(t, func() bool {
+		_, _ = chain.Request(context.Background(), req)
+		prefixesUpdated.RLock()
+		defer prefixesUpdated.RUnlock()
+		v := prefixesUpdated.b
+		return v
+	}, time.Second, time.Millisecond*100)
+}
+
+func TestUniqueRequestPrefixes(t *testing.T) {
+	prefixes := []string{"172.16.1.0/24", "10.32.0.0/12", "10.96.0.0/12", "10.20.128.0/17", "10.20.64.0/18", "10.20.8.0/21", "10.20.4.0/22"}
+	testConfig := strings.Join(append([]string{"prefixes:"}, prefixes...), "\n- ")
+	reqPrefixes := []string{"100.1.1.0/13", "10.32.0.0/12", "10.96.0.0/12", "10.20.0.0/24", "10.20.128.0/17", "10.20.64.0/18", "10.20.16.0/20", "10.20.2.0/23"}
+
+	configPath := path.Join(os.TempDir(), "excluded_prefixes.yaml")
+	f, err := os.Create(configPath)
+	require.Nil(t, err)
+	defer func() { _ = os.Remove(configPath) }()
+	_, err = f.WriteString(testConfig)
+	require.Nil(t, err)
+	err = f.Close()
+	require.Nil(t, err)
+
+	chain := next.NewNetworkServiceServer(NewServerFromPath(configPath), checkrequest.NewServer(t, func(t *testing.T, request *networkservice.NetworkServiceRequest) {
+		require.True(t, uniquePrefixes(request.Connection.Context.IpContext.ExcludedPrefixes))
```

Comment:
I think this function is not needed we can just check that actual value is equal to `expected const value`.

Reply:
ok, i will take it into account